crowbar_framework/app/models/service_object.rb

```diff
@@ -1826,4 +1826,35 @@ def skip_unready_nodes(bc, inst, new_elements, old_elements)
     logger.debug("skip_unready_nodes: exit for #{bc}:#{inst}")
     [cleaned_elements, pre_cached_nodes]
   end
+
+  # return true if the new attributes are different from the old ones
+  def node_changed_attributes?(node, old_role, new_role)
+    if old_role.default_attributes[@bc_name] != new_role.default_attributes[@bc_name]
+      logger.debug("#{@bc_name} skip_batch_for_node?: not skipping #{node} (attribute change)")
```

Comment:
@vuntz I think it's better to remove any debug information from here and from the method node_changed_roles and move them to each implementation of skip_unchanged_node?
What do you think?